File: evals/workflow_value/results/2026-09-25/cells/05/submitted-project/src/power_cpp_monitor/test/test_conversion.cpp
#include <gtest/gtest.h>

#include <cmath>

#include "power_cpp_monitor/conversion.hpp"

TEST(Conversion, ReturnsWattsWhenValid)
{
  power_interfaces::msg::PowerReading msg;
  msg.power_w = 12.5;
  msg.valid = true;
  EXPECT_DOUBLE_EQ(power_cpp_monitor::normalize(msg), 12.5);
}

TEST(Conversion, ReturnsNanWhenInvalid)
{
  power_interfaces::msg::PowerReading msg;
  msg.power_w = 12.5;
  msg.valid = false;
  EXPECT_TRUE(std::isnan(power_cpp_monitor::normalize(msg)));
}

File: evals/workflow_value/results/2026-09-25/cells/06/submitted-project/src/power_py_monitor/test/test_conversion.py
import math

from power_interfaces.msg import PowerReading

from power_py_monitor.conversion import normalize


def test_valid_reading_returns_watts():
    msg = PowerReading()
    msg.power_w = 12.5
    msg.valid = True
    assert normalize(msg) == 12.5


def test_invalid_reading_returns_nan():
    msg = PowerReading()
    msg.power_w = 12.5
    msg.valid = False
    assert math.isnan(normalize(msg))
